packages/flutter/lib/src/services/raw_keyboard_macos.dart

```diff
@@ -0,0 +1,222 @@
+  ///
+  ///   * [Apple's NSEvent documentation](https://developer.apple.com/documentation/appkit/nsevent/1524605-charactersignoringmodifiers?language=objc)
+  final String charactersIgnoringModifiers;
+
+  /// The virtual key code for the keyboard key associated with a key event.
+  ///
+  /// See also:
+  ///
+  ///   * [Apple's NSEvent documentation](https://developer.apple.com/documentation/appkit/nsevent/1534513-keycode?language=objc)
+  final int keyCode;
+
+  /// A mask of the current modifiers using the values in Modifier Flags.
+  /// 
+  /// See also:
+  ///
+  ///   * [Apple's NSEvent documentation](https://developer.apple.com/documentation/appkit/nsevent/1535211-modifierflags?language=objc)
+  final int modifiers;
+
+  @override
+  String get keyLabel => charactersIgnoringModifiers.isEmpty ? null : charactersIgnoringModifiers;
+
+  @override
+  PhysicalKeyboardKey get physicalKey => kMacOsToPhysicalKey[keyCode] ?? PhysicalKeyboardKey.none;
+
+  @override
+  LogicalKeyboardKey get logicalKey {
+    // Look to see if the keyCode is a printable number pad key, so that a
+    // difference between regular keys (e.g. "=") and the number pad version
+    // (e.g. the "=" on the number pad) can be determined.
+    final LogicalKeyboardKey numPadKey = kMacOsNumPadMap[keyCode];
+    if (numPadKey != null) {
+      return numPadKey;
+    }
+
+    // Look to see if the keyCode is one we know about and have a mapping for.
+    if (keyLabel != null &&
+        !LogicalKeyboardKey.isControlCharacter(keyLabel) &&
+        charactersIgnoringModifiers.isNotEmpty) {
+      final int keyId = LogicalKeyboardKey.unicodePlane |
+        (charactersIgnoringModifiers.codeUnitAt(0) & LogicalKeyboardKey.valueMask);
+      return LogicalKeyboardKey.findKeyByKeyId(keyId) ?? LogicalKeyboardKey(
+        keyId,
+        keyLabel: keyLabel,
+        debugName: kReleaseMode ? null : 'Key ${keyLabel.toUpperCase()}',
+      );
+    }
+
+    // This is a non-printable key that we don't know about, so we mint a new
+    // code with the autogenerated bit set.
+    const int macOsKeyIdPlane = 0x00500000000;
+
+    // Keys like "backspace" won't have a character, but it's known by the physical keyboard.
+    // Since there is no logical keycode map for macOS (macOS uses the keycode to reference
+    // physical keys), a LogicalKeyboardKey is created with the physical key's HID usage and
+    // debugName. This avoids the need of duplicating the physical keys map.
+    if (physicalKey != PhysicalKeyboardKey.none) {
+      return LogicalKeyboardKey(
+        physicalKey.usbHidUsage,
+        keyLabel: physicalKey.debugName,
+        debugName: physicalKey.debugName,
+      );
+    }
+
+    return LogicalKeyboardKey(
+      macOsKeyIdPlane | keyCode | LogicalKeyboardKey.autogeneratedMask,
+      debugName: kReleaseMode ? null : 'Unknown macOS key code $keyCode',
+    );
+  }
+
+  @override
+  bool isModifierPressed(ModifierKey key, {KeyboardSide side = KeyboardSide.any}) {
```

Comment:
You should be able to differentiate which side the modifier key was on for shift, control, option, and meta.

Maybe [this Stack Overflow answer](https://stackoverflow.com/questions/10715910/is-there-a-way-to-differentiate-between-left-and-right-shift-keys-being-pressed) helps?

Reply:
Thanks! Thought the left and right modifiers would be documented with the modifier flags, but seems like they're not :(